src/ai.hpp

```diff
@@ -21,11 +21,11 @@ struct PlayerOutcome
 class AI
 {
 public:
-	const int MaxPly = 5;
+	static const int MaxPly = 3;
```

Comment:
This was the source of the bug, somehow. Apparently, if you use a non-static constant, C++ just randomly assigns a value of it, so the MaxPly was essentially undefined and thus one of the AI base cases was never fulfilled.

Reply:
How'd you figure this one out?

Reply:
I noticed that the value of MaxPly was a hugely negative number whose exact value changed each time I ran the program. It took ages to track it to that point but I realized then that it wasn't retrieving the constant value.

Most confusingly is that VillainSpawnProbability constant seems to work despite its similar declaration.